src/FileDownloader.ts

```diff
@@ -113,20 +109,26 @@ export default class FileDownloader implements IFileDownloader {
         }
 
         if (cancellationToken?.isCancellationRequested ?? false) {
-            await rimrafAsync(tempFileDownloadPath, rimrafOptions);
+            await rimrafAsync(tempFileDownloadPath);
             throw new DownloadCanceledError();
         }
 
-        // If the file/folder already exists, remove it now
-        await rimrafAsync(fileDownloadPath, rimrafOptions);
+        try {
+            // If the file/folder already exists, remove it now
+            await rimrafAsync(fileDownloadPath);
```

Comment:
With this change, we're not retrying anymore the rimraf call. Is this a concern, or do you expect the EPERM issue to be fixed by the rename call below anyway?

Reply:
From the readme of the rimraf package:
```
**maxBusyTries**

If an EBUSY, ENOTEMPTY, or EPERM error code is encountered on Windows systems, then rimraf will retry with a linear backoff wait of 100ms longer on each try. The default maxBusyTries is 3.

Only relevant for async usage.

**emfileWait**

If an EMFILE error is encountered, then rimraf will retry repeatedly with a linear backoff of 1ms longer on each try, until the timeout counter hits this max. The default limit is 1000.
```

So, I believe they are already retrying and I think most of EPERM errors are because of the rename.

Reply:
OK sounds good, thanks for the clarification!